ddl/placement/utils.go

```diff
@@ -131,3 +131,26 @@ func BuildPlacementCopyBundle(oldBundle *Bundle, newID int64) *Bundle {
 	}
 	return newBundle
 }
+
+// GetLeaderDCByBundle return the leader's dc by Bundle
+func GetLeaderDCByBundle(bundle *Bundle, dcLabelKey string) string {
+	dc := getRoleDCByBundle(bundle, dcLabelKey, Leader)
+	if len(dc) > 0 {
+		return dc
+	}
+	// If there is placement policy for leader, fallback to search voter
+	return getRoleDCByBundle(bundle, dcLabelKey, Voter)
+}
+
+func getRoleDCByBundle(bundle *Bundle, dcLabelKey string, role PeerRoleType) string {
+	for _, rule := range bundle.Rules {
+		if rule.Role == role {
+			for _, cons := range rule.LabelConstraints {
+				if cons.Key == dcLabelKey && len(cons.Values) > 0 {
+					return cons.Values[0]
```

Comment:
Cannot choose anyone from `Values`.
e.g.
```sql
CONSTRAINTS='["+zone=bj","+zone=sh"]' ROLE=leader
```
The leader could be scheduled in either `bj` or `sh`. In this case, it's not possible to visit this partition in a local transaction.